File: Difficulty: Easy/Intersection of two sorted arrays/intersection-of-two-sorted-arrays.py
#User function Template for python3


class Solution:
    #Function to return a list containing the intersection of two arrays.
    def intersection(self, arr1, arr2):
        res=[]
        n1=len(arr1)
        n2=len(arr2)
        i,j=0,0
        while i<n1 and j<n2 :
            if i>0 and arr1[i-1]==arr1[i]:
                i+=1
                continue
            if j>0 and arr2[j-1]==arr2[j]:
                j+=1
                continue
            if arr1[i]<arr2[j]:
                i+=1
            elif arr1[i]>arr2[j]:
                j+=1
            else:
                res.append(arr1[i])
                i+=1
                j+=1
        return res



#{ 
 # Driver Code Starts
#Initial Template for Python 3
import io
import sys

if __name__ == '__main__':
    test_cases = int(input())
    for cases in range(test_cases):
        arr1 = list(map(int, input().strip().split()))
        arr2 = list(map(int, input().strip().split()))
        ob = Solution()
        l = ob.intersection(arr1, arr2)
        if (len(l) > 0):
            print(*l)
        else:
            print("[]")
        print("~")

# } Driver Code Ends

File: Difficulty: Easy/Intersection of Two Sorted Arrays/intersection-of-two-sorted-arrays.java
class Solution {
    ArrayList<Integer> intersection(int[] a, int[] b) {
        // code here
        HashSet<Integer> hs=new HashSet<>();
        for(int num:a){
            hs.add(num);
        }
        HashSet<Integer> hs2=new HashSet<>();
        ArrayList<Integer> res=new ArrayList<>();
        for(int num:b){
            if(hs.contains(num) && !hs2.contains(num)){
                res.add(num);
                hs2.add(num);
            }
        }
        return res;
    }
}